### PR TITLE
vagrant: add Vagrantfile and some basic configuration

### PR DESCRIPTION
**Pull request by ffrank**

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,43 @@
+
```
Comment (roidelapluie) [inline Vagrantfile]: newlines?

File: vagrant/bashrc.mgmt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+
```
Comment (roidelapluie) [inline vagrant/bashrc.mgmt]: newline?

File: vagrant/bashrc.mgmt [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+# alias for simple access
+alias cdrepo='cd $GOPATH/src/github.com/purpleidea/mgmt'
+
+export GOPATH=~/go
```
Comment (roidelapluie) [inline vagrant/bashrc.mgmt]: I would call this file mgmt.bashrc, not bashrc.mgmt

File: misc/make-deps.sh [hunk 4]
```diff
@@ -2,6 +2,8 @@
 # setup a simple go environment
 XPWD=`pwd`
 ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && cd .. && pwd )"	# dir!
+VAGRANT_MODE=
```
Comment (purpleidea) [inline misc/make-deps.sh]: You should have this ="" for example. So bash with unset var warnings won't happen IIRC.

File: vagrant/bashrc.mgmt (hunk 3 above)
Comment (purpleidea) [inline vagrant/bashrc.mgmt]: Can you change to 'gopath' please.

File: vagrant/motd [hunk 5]
```diff
@@ -0,0 +1,14 @@
+Welcome to the mgmt hacking VM.
+
+Please make sure that your authorized key for GitHub is added to the SSH agent
+on the host machine (see `ssh-add -l`).
+
+You can jump into the mgmt repo with the alias `cdrepo`.
+
+To get started, try:
+
+  cdrepo
```
Comment (purpleidea) [inline vagrant/motd]: cute idea. Nit: let's name this cdmgmt if you want to have an alias.

Comment (ffrank) [inline vagrant/motd]: Hah check. That is actually the alias I use on my laptop :fist_right: :fist_left:

**Comment by purpleidea** [comment]
@ffrank I'm under the impression that in the vagrant environment *everything* should run as root to make everyone's life easier...

**Comment by ffrank** [comment]
I kind of like that you can run unprivileged by default, so that you get some feedback on what does and does not Just Work. Of course, in the absence of DBUS and the ability to get root ad hoc for resources like `pkg`, this is less fun in the vagrant environment.

OK, I'll probably go for root then.

**Comment by purpleidea** [comment]
@ffrank I actually don't disagree, I was more trying to avoid making the make-deps script more complicated. Up to you!

**Comment by purpleidea** [comment]
... I'd actually rather just duplicate some of the deps code in the Vagrantfile directly, rather than go crazy about code reuse of make-deps.sh

**Comment by ffrank** [comment]
Good point, I was thinking about how to do this best...and yes, complexity is not ideal. But I really want to avoid repitition.
First idea was to rip the make-deps.sh into two scripts, and use both from `make deps`, but maybe only one for vagrant. That might be cleaner after all.

**Comment by purpleidea** [comment]
@ffrank Just ping me when ready for a new review since GH is lame and doesn't notify when patch is updated.

**Comment by ffrank** [comment]
Pushed a rebased and adapted version, incorporating all the feedback. Thanks for that.

I still build the environment for the vagrant user. root would just feel weird for coding and interacting with github.

**Comment by purpleidea** [comment]
@ffrank Thanks for the update! I know a few folks who are excited to have this upstream!

A few things... We already have a commit message check, sorry: https://github.com/purpleidea/mgmt/blob/master/test/test-commit-message.sh (patches welcome if you think it needs improvements) -- Actually it only works on travis, so adding the `git log` fallback would be a welcome patch (in a separate commit)

Secondly, as per: https://github.com/purpleidea/mgmt/pull/132#issuecomment-278908556 I think it's better to keep the make-deps.sh script as is, and just copy the relevant parts into the vagrantfile if needed. I think splitting this up isn't necessary for now, sorry.

**Comment by ffrank** [comment]
Sorry, the commit check was WiP I abandoned when I realized @roidelapluie had built that already. Added it to this commit on accident.

Should be fine now.

**Comment by purpleidea** [comment]
Rebased and merged!

**Pull request closed without merging**